src/App.jsx

```diff
@@ -13,34 +14,95 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_MODE_ALPHABET = 'alphabet';
+const SORT_MODE_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function setSort(goods, sortType, isReverse) {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortType) {
+    case SORT_MODE_ALPHABET:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SORT_MODE_LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortMode, setSortMode] = useState('');
```

Comment:
The initial state of `sortMode` is set to an empty string. This can lead to unexpected behavior since the `setSort` function does not handle an empty string as a valid sort type. Consider setting a default sort mode, such as `SORT_MODE_ALPHABET` or `SORT_MODE_LENGTH`, to ensure consistent behavior.